src/modules/Builder/features/ActivityItems/ActivityItems.tsx

```diff
@@ -75,20 +68,22 @@ export const ActivityItems = () => {
     for (const itemIndex of indexListToTrigger) {
       trigger(`${itemsName}.${itemIndex}`);
     }
+
     firstSystemIndex === -1 ? appendItem(item) : insertItem(firstSystemIndex, item);
-    setActiveItemIndex(firstSystemIndex === -1 ? items?.length : firstSystemIndex);
+
+    navigateToItem(item);
   };
 
   const handleInsertItem = (index: number, item?: ItemFormValues) => {
-    const itemToInsert = item ?? getNewActivityItem();
+    const itemToInsert = (item ?? getNewActivityItem()) as ItemFormValues;
```

Comment:
if it won't affect a lot of places could you add `ItemFormValues` type for result of getNewActivityItem()?